generic/utils/ras.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Pavithra <pavrampu@linux.vnet.ibm.com>
+
+import os
+import commands
+from shutil import copyfile
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class RASTools(Test):
+
+    '''
+    This test verifies below RAS tools:
+    set_poweron_time - set_poweron_time schedules the power on time
+    sys_ident - sys_ident provides unique system identification information
+    lsmcode - lsmcode provides FW version information
+    drmgr - drmgr can be used for pci, cpu or memory hotplug
+    lsprop - lsprop provides device tree information
+    lsslot - lsslot lists the slots based on the option provided
+    lsvio - lsvio lists the virtual I/O adopters and devices
+    nvram - nvram command retrieves and displays NVRAM data
+    ofpathname - ofpathname translates the device name between logical name and Open Firmware name
+    ppc64_cpu - ppc64_cpu is used to set cpu options
+    rtas_ibm_get_vpd - rtas_ibm_get_vpd gives vpd data
+    rtas_errd and rtas_dump - rtas_errd adds RTAS events to /var/log/platform and rtas_dump dumps RTAS events
+    '''
+    is_fail = 0
+
+    def run_cmd(self, cmd):
+        global is_fail
+        print "Output of command \"%s\":" % cmd
+        cmd_result = process.run(cmd, ignore_status=True, sudo=True)
+        if cmd_result.exit_status != 0:
+            self.is_fail = 1
+        return
+
+    def setUp(self):
+        architecture = os.uname()[4]
+        if "ppc" not in architecture:
+            self.skip("supported only on Power platform")
+        sm = SoftwareManager()
+        for package in ("ppc64-diag", "powerpc-utils", "lsvpd"):
+            if not sm.check_installed(package) and not sm.install(package):
+                self.error(
+                    "Fail to install %s required for this test." % package)
+        log_dir = os.path.join(self.outputdir, 'log')
+        if not os.path.exists(log_dir):
+            os.makedirs(log_dir)
+
+    def test1_set_poweron_time(self):
+        print "===============Executing set_poweron_time tool test==============="
+        self.run_cmd("set_poweron_time -m")
+        self.run_cmd("set_poweron_time -h")
+        self.run_cmd("set_poweron_time -d m2")
+        self.run_cmd("set_poweron_time -t M6D15h12")
+        if self.is_fail == 1:
+            self.fail("RAS check failed")
```

Comment:
If you decide to use my recommendation, this would have to become:

```python
if self.is_fail:
    self.fail("%s RAS check(s) failed" % self.is_fail)
```
I think that would be a useful info :wink: